Assets/Vimeo/Scripts/Player/VimeoPlayer.cs

```diff
@@ -223,17 +224,41 @@ private void VideoControllerPlayVideo()
                 
                 if (videoPlayerType == VideoPlayerType.AVProVideo && mediaPlayer != null) {
                     mediaPlayer.OpenVideoFromFile(RenderHeads.Media.AVProVideo.MediaPlayer.FileLocation.AbsolutePathOrURL, file_url, autoPlay || playVideoAfterLoad);
-                }
+                } 
+#endif // VIMEO_AVPRO_VIDEO_SUPPORT
 #if VIMEO_DEPTHKIT_SUPPORT
-                // TODO
-                else if (videoPlayerType == VideoPlayerType.DepthKit && depthKitClip != null) {
-                    // depthKitClip._moviePath = file_url;
-                    // depthKitClip._metaDataText = vimeoVideo.description;
-                    // depthKitClip.GetComponent<RenderHeads.Media.AVProVideo.MediaPlayer>().OpenVideoFromFile(RenderHeads.Media.AVProVideo.MediaPlayer.FileLocation.AbsolutePathOrURL, file_url, autoPlay);
-                    // depthKitClip.RefreshRenderer();
+                if (videoPlayerType == VideoPlayerType.Depthkit && depthKitClip != null) {
+#if VIMEO_AVPRO_VIDEO_SUPPORT   
+                    if (depthKitClip.gameObject.GetComponent<RenderHeads.Media.AVProVideo.MediaPlayer>() != null){
+                        depthKitClip.gameObject.GetComponent<RenderHeads.Media.AVProVideo.MediaPlayer>().OpenVideoFromFile(RenderHeads.Media.AVProVideo.MediaPlayer.FileLocation.AbsolutePathOrURL, file_url, autoPlay);
+                    }
+#endif // VIMEO_AVPRO_VIDEO_SUPPORT
+                    if (depthKitClip.gameObject.GetComponent<VideoPlayer>() != null) {
+                        if (this.selectedResolution != StreamingResolution.Adaptive) {
+                            depthKitClip.gameObject.GetComponent<VideoPlayer>().url = vimeoVideo.GetVideoFileUrlByResolution(selectedResolution);
+                        } else {
+                            Debug.LogError("[Vimeo] Unity video player does not support adaptive video try selecting a specific quality in the Vimeo Player or use AVPro video on the Depthkit clip");
+                        }
+                    }
+
+                    JSONNode metadata = vimeoVideo.GetMetadata();
+
+                    if (HasValidDepthkitMetadata(metadata)) {
+                        depthKitClip._metaDataFile = new TextAsset(metadata.ToString());
+                        depthKitClip._needToRefreshMetadata = true;   
+                        if (OnVideoStart != null) {
+                            OnVideoStart();
```

Comment:
This is technically hacking an event trigger. `OnVideoStart` in default context it means the video just started being rendered. But in this context, it means the Depthkit metadata has been loaded, right? We should aim to be consistent.